src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java

```diff
@@ -247,33 +267,13 @@ private static boolean isStreamingPolicyEnabled(
 
     public void createKustoIngestClient(KustoSinkConfig config) {
         try {
-            if (!Strings.isNullOrEmpty(config.getAuthAppid())) {
-                if (Strings.isNullOrEmpty(config.getAuthAppkey())) {
-                    throw new ConfigException("Kusto authentication missing App Key.");
-                }
+            ConnectionStringBuilder ingestConnectionStringBuilder = createKustoEngineConnectionString(config, config.getKustoIngestUrl());

```

Comment:
Use the generic method instead of duplicating code. Here we use the ingest URL, and below we use the `getKustoEngineUrl` for the streaming client